apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data/SpannerTransaction.cs

```diff
@@ -563,5 +570,10 @@ private void CheckCompatibleMode(TransactionMode mode)
                     throw new ArgumentOutOfRangeException(nameof(mode), mode, null);
             }
         }
+
+        private static readonly TimeSpan s_maxCommitDelay = TimeSpan.FromMilliseconds(500);
```

Comment:
I'm somewhat nervous of building this into the client. I think it's absolutely fine to check that it's not negative, but I can easily see the Spanner team increasing this over time, and the libraries getting out of date.

Effectively, I'd say that "not negative" is a *logical* validation, whereas "at most 500ms" is an *implementation-specific* validation.

Happy to be persuaded though :)